code/controllers/subsystem/lighting.dm

```diff
@@ -32,62 +32,55 @@ var/datum/subsystem/lighting/SSlighting
 /datum/subsystem/lighting/fire()
 	changed_lights_workload = MC_AVERAGE(changed_lights_workload, changed_lights.len)
 
-//	for(var/area/A in sortedAreas)
-//		A.luminosity = 1
-
-	for(var/datum/light_source/thing in changed_lights)
-		thing.check()
+	for(var/thing in changed_lights)
+		var/datum/light_source/LS = thing
+		LS.check()
 	changed_lights.Cut()
 
 	changed_turfs_workload = MC_AVERAGE(changed_turfs_workload, changed_turfs.len)
-	for(var/turf/thing in changed_turfs)
-		if(thing && thing.lighting_changed)
-			thing.redraw_lighting()
+	for(var/thing in changed_turfs)
+		var/turf/T = thing
+		if(T.lighting_changed)
+			T.redraw_lighting()
 	changed_turfs.Cut()
 
-//	for(var/area/A in sortedAreas)
-//		A.luminosity = !A.lighting_use_dynamic
-
 //same as above except it attempts to shift ALL turfs in the world regardless of lighting_changed status
 //Does not loop. Should be run prior to process() being called for the first time.
 //Note: if we get additional z-levels at runtime (e.g. if the gateway thin ever gets finished) we can initialize specific
 //z-levels with the z_level argument
 /datum/subsystem/lighting/Initialize(timeofday, z_level)
 
-//	for(var/area/A in sortedAreas)
-//		A.luminosity = 1
 
-	for(var/datum/light_source/thing in changed_lights)
-		thing.add_effect()
-		thing.changed = 0
+	for(var/thing in changed_lights)
+		var/datum/light_source/LS = thing
+		LS.check()
```

Comment:
This causes problems when you initialize arbitrary z-levels while existing lights need to be updated